_posts/2024-08-19-credential-store-for-encrypted-expressions.adoc

```diff
@@ -0,0 +1,136 @@
+
+This is where encrypted expressions are useful. Support for encrypted expressions allow us to specify sensitive information, like the password for a credential-store without using plain-text values. Encrypted expressions can also be used for system properties, where a value expression is specified. 
+And lastly, credential-store can only be used for credential-reference attributes. But there are other attributes where we can enter sensitive information, but it does not include a credential reference. This guide will explain the these three use cases for encrypted expressions and how to configure them. 
+
+== Configuring Encrypted Expressions
+Let's first create a `secret-key-credential-store` that we can use to store the encrypted expressions. Just like a regular credential-store file, this will also create a .cs file inside the directory you specify. However, this will be human readable. 
+```
+/subsystem=elytron/secret-key-credential-store=newCredStore:add(path=newSecretKeyCred.cs, relative-to=jboss.server.config.dir)
+```
+If you already had a .cs file that you would like to use, then you can specify the path here and set the `create` and `populate` attributes to `false`. These attributes are `true` by default as seen here: 
+```
+{
+    "outcome" => "success",
+    "result" => {
+        "path" => "newSecretKeyCred.cs",
+        "relative-to" => "jboss.server.config.dir",
+        "create" => true,
+        "default-alias" => "key",
+        "key-size" => 256,
+        "populate" => true
+    }
+}
+``` 
+Here is an example of the contents of the newSecretKeyCred.cs file: 
+```
+# Properties Credential Store (Do Not Modify)
+key=RUxZAUsOhzQhCv7KbD9f2SGBgWkmTC9l0lgu28FWM4UfRBY7QQ==
+```
+As you can see there has been a key created for you already using the default alias `key`. You can create a new key with a new alias of your own using the following command: 
+```
+/subsystem=elytron/secret-key-credential-store=newCredStore:generate-secret-key(alias=newSecretKey, key-size=128)
+```
+Now let's use these keys for the use cases mentioned above. 
+
+== Encrypted Expression for Credential Reference 
+Now that we have a generated secret key in a credential store we can activate the resource responsible for handling encrypted expressions. And we can use that to convert a clear-text password phrase into an encrypted expression which will be stored in the credential-store. 
```

Comment:
remove "which will be stored in the credential-store".

Reply:
(the secret key from the store is used to encrypt a clear text password and the encrypted expression can be decrypted using the key from the store. the encrypted expression itself isn't stored in the secret key credential store)

Reply:
Fixed! 
Thanks @fjuma